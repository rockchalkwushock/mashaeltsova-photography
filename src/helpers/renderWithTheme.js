import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../utils/theme'

/**
 * @fileOverview
 *
 * HOC for testing components with `theme` prop being used.
 */

const renderWithTheme = Component =>
  mount(<ThemeProvider theme={theme}>{Component}</ThemeProvider>)

export default renderWithTheme
