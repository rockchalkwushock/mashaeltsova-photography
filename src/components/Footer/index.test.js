import React from 'react'

import { renderWithTheme } from '../../helpers'
import AppFooter from './'

describe('Component: <AppFooter />', () => {
  let wrapper
  beforeAll(() => {
    wrapper = renderWithTheme(<AppFooter />)
  })
  test('should render markup', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
