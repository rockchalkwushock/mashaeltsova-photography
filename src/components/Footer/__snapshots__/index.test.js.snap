// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <AppFooter /> should render markup 1`] = `
.c0 {
  background-color: #333;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-top: 2em;
  text-align: center;
  width: 100vw;
}

.c1 {
  text-align: center;
}

.c2 {
  text-align: center;
}

@media (min-width:736px) {
  .c0 {
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    padding: 2em 1em 1em 1em;
  }
}

@media (min-width:736px) {
  .c1 {
    text-align: left;
  }
}

@media (min-width:736px) {
  .c2 {
    text-align: right;
  }
}

<ThemeProvider
  theme={
    Object {
      "sizes": Object {
        "lg": "1000px",
        "md": "736px",
        "sm": "540px",
      },
    }
  }
>
  <AppFooter>
    <Footer>
      <footer
        className="c0"
      >
        <FooterContainer
          left={true}
        >
          <div
            className="c1"
          >
            <h2>
              Left
            </h2>
          </div>
        </FooterContainer>
        <FooterContainer>
          <div
            className="c2"
          >
            <h2>
              Right
            </h2>
          </div>
        </FooterContainer>
      </footer>
    </Footer>
  </AppFooter>
</ThemeProvider>
`;
